dwave/system/composites/tiling.py

```
@@ -104,73 +123,143 @@ def __init__(self, sampler, sub_m, sub_n, t=4):
+                if len(sampler.properties['topology']['shape']) != 3: 
+                    raise ValueError('topology shape is not of length 3 '
+                                     '(not compatible with chimera)')
+                if sampler.properties['topology']['shape'][2] != t: 
+                    raise ValueError('Tiling methodology requires that solver'
+                                     'and subproblem have identical shore size')
+                m = sampler.properties['topology']['shape'][0]
+                n = sampler.properties['topology']['shape'][1]
+            else:
+                if len(sampler.properties['topology']['shape']) != 1: 
+                    raise ValueError('topology shape is not of length 1 '
+                                     '(not compatible with pegasus)')
+                #Full yield in odd-couplers also required.
+                #Generalizes chimera subgraph requirement and leads to some 
+                #simplification of expressions, but at with a cost in cell-yield
+                edges_per_cell += t
+                #Square solvers only by pegasus lattice definition PN yields
+                #3 by N-1 by N-1 cells:
+                num_sublattices=3
+                m = n = sampler.properties['topology']['shape'][0] - 1
+                if t!=4:
+                    raise ValueError(
+                        't=4 for all pegasus processors, value is not typically'
+                        'stored in solver properties and is difficult to infer.'
+                        'Therefore only the value t=4 is supported.')
+                    
+        else:
+            warnings.warn("Incomplete solver topology information."
+                          "Falling back to unsafe legacy branch.",
+                          DeprecationWarning)
+            # assume square lattice shape, and high yield, chimera system
+            m = n = int(ceil(sqrt(ceil(len(sampler.structure.nodelist) / nodes_per_cell))))  
```

@jackraymond, isn't this calculation off for Pegasus so if the `shape` info is missing from a Pegasus sampler then the `m` here is for Chimera, and using it in line 187 (`system = dnx.pegasus_graph(m, ...`) below will fail?

---

I wanted to keep the original code intact, since it works (except in pathological cases) for Chimera. This deprecation warning is probably not a strong enough disincentive. Better to just throw an error given that Pegasus use case will be increasingly common

---

In other words, deprecate rather than warn.